### PR TITLE
Add efs capability

### PR DESCRIPTION
**Pull request by sparrc**
just adding the "efs" capability to the feature branch.

### CODE REVIEW

File: agent/app/agent_capability_unix.go [hunk 1]
```diff
@@ -153,6 +153,10 @@ func (agent *ecsAgent) appendFirelensFluentbitCapabilities(capabilities []*ecs.A
 	return appendNameOnlyAttribute(capabilities, attributePrefix+capabilityFirelensFluentbit)
 }
 
+func (agent *ecsAgent) appendEFSCapabilities(capabilities []*ecs.Attribute) []*ecs.Attribute {
```
Comment (sharanyad) [inline agent/app/agent_capability_unix.go]: missing calling of this method

Comment (sharanyad) [inline agent/app/agent_capability_unix.go]: needs to be there in `agent_capability_windows.go` as well

Comment (sparrc) [reply agent/app/agent_capability_unix.go]: I thought we are only supporting linux for now?

Comment (sharanyad) [reply agent/app/agent_capability_unix.go]: oh yes! just a dummy method returning attributes.

Comment (sparrc) [reply agent/app/agent_capability_unix.go]: I see now, pushed

**Comment by adnxn** [comment]
same changes as this? https://github.com/aws/amazon-ecs-agent/pull/2119

nit: mb remove those redundant test cases as well?

**Comment by sparrc** [comment]
@adnxn yes that looks related but this is the most up-to-date capability, confirmed with CP.

AFAICT that separated EIA unit test has been removed already in some other PR

File: agent/app/agent_capability.go [hunk 2]
```diff
@@ -185,6 +186,9 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 	// support aws router capabilities for log driver router
 	capabilities = agent.appendFirelensLoggingDriverCapabilities(capabilities)
 
+	// support efs on ecs capabilities
+	capabilities = agent.appendEFSCapabilities(capabilities)
```
Comment (fenxiong) [inline agent/app/agent_capability.go]: nit: update comment above capabilities() to include new one

Comment (sparrc) [reply agent/app/agent_capability.go]: to include the new one of what?

Comment (fenxiong) [reply agent/app/agent_capability.go]: new capabiliity "ecs.capability.efs"

Comment (sparrc) [reply agent/app/agent_capability.go]: got it, pushed

File: agent/app/agent_capability_unix.go (hunk 1 above)
Comment (sparrc) [reply agent/app/agent_capability_unix.go]: done

**Pull request merged**